src/button-card.ts

```diff
@@ -0,0 +1,569 @@
+    let colorValue: string = '';
+    let color: undefined | string = undefined;
+    if (configState && configState.color) {
+      colorValue = configState.color;
+    } else {
+
+      if (this.config!.color != 'auto' && state && state.state == 'off') {
+        colorValue = this.config!.color_off;
+      } else {
+        if (this.config!.color) {
+          colorValue = this.config!.color;
+        }
+      }
+    }
+    if (colorValue == 'auto') {
+      if (state) {
+        if (state.attributes.rgb_color) {
+          color = `rgb(${state.attributes.rgb_color.join(',')})`
+        } else {
+          color = this.getDefaultColorForState(state)
+        }
+      } else {
+        color = this.config!.default_color;
+      }
+    } else {
+      if (!colorValue) {
+        if (state) {
+          color = this.getDefaultColorForState(state)
+        } else {
+          color = this.config!.default_color;
+        }
+      } else {
+        color = colorValue;
+      }
+    }
+    return color;
+  }
+
+  private buildIcon(state: HassEntity | null, configState: StateConfig | undefined) {
+    let icon: undefined | string = undefined;
+    if (configState && configState.icon) {
+      icon = configState.icon;
+    } else if (this.config!.icon) {
+      icon = this.config!.icon;
+    } else {
+      if (state && state.attributes) {
+        icon = state.attributes.icon ?
+          state.attributes.icon :
+          domainIcon(helpers.computeDomain(state.entity_id), state.state);
+      }
+    }
+    return icon;
+  }
+
+  private buildStyle(state: HassEntity | null, configState: StateConfig | undefined) {
+    let cardStyle = '';
+    let styleArray: CssStyleConfig | undefined = undefined;
+    if (state) {
+      if (configState && configState.style) {
+        styleArray = configState.style;
+      } else if (this.config!.style) {
+        styleArray = this.config!.style;
+      }
+    } else {
+      if (this.config!.style)
+        styleArray = this.config!.style;
+    }
+    if (styleArray) {
+      styleArray.forEach((cssObject) => {
+        const attribute = Object.keys(cssObject)[0];
+        const value = cssObject[attribute];
+        cardStyle += `${attribute}: ${value}; `;
+      });
+    }
+    return cardStyle;
+  }
+
+  private buildName(state: HassEntity | null, configState: StateConfig | undefined) {
+    if (this.config!.show_name === false) {
+      return null;
+    }
+    let name: string | null = null;
+    if (configState && configState.name) {
+      name = configState.name;
+    } else if (this.config!.name) {
+      name = this.config!.name;
+    } else {
+      if (state) {
+        name = state.attributes && state.attributes.friendly_name ?
+          state.attributes.friendly_name : helpers.computeEntity(state.entity_id);
+      }
+    }
+    return name;
+  }
+
+  private buildStateString(state: HassEntity | null) {
```

Comment:
```suggestion
  private buildStateString(state: HassEntity | undefined): string | undefined {
```
Always use `undefined` and not `null` per the [TypeScript spec](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined)